File: ocaml/space-age/space_age.ml
type planet = Mercury | Venus | Earth | Mars
            | Jupiter | Saturn | Neptune | Uranus

let age_on p a = 
  let a' = float_of_int a /. float_of_int 31557600 in
  match p with
  | Mercury -> a' /. 0.2408467
  | Venus -> a' /. 0.61519726
  | Earth -> a'
  | Mars -> a' /. 1.8808158 
  | Jupiter -> a' /. 11.862615
  | Saturn -> a' /. 29.447498
  | Uranus -> a' /. 84.016846
  | Neptune -> a' /. 164.79132

File: fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet =
  | Earth
  | Mercury
  | Venus
  | Mars
  | Jupiter
  | Saturn
  | Uranus
  | Neptune

let age (planet: Planet) (seconds: int64): float =
  let helper factor = System.Math.Round ((float seconds / 31557600.0 / factor), 2)
  match planet with
  | Earth -> helper 1.0
  | Mercury -> helper 0.2408467
  | Venus -> helper 0.61519726
  | Mars -> helper 1.8808158
  | Jupiter -> helper 11.862615
  | Saturn -> helper 29.447498
  | Uranus -> helper 84.016846
  | Neptune -> helper 164.79132
